src/framework/tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,134 @@
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+import ReactDOM from 'react-dom';
+
+import {
+  Box,
+  Text,
+} from '../../framework/framework';
+
+export default class Tooltip extends Component {
+  constructor(props) {
+    super(props);
+
+    this.content = this.props.message;
+    this.children = this.props.children;
+    this.onMouseEnter = this.onMouseEnter.bind(this);
+    this.onMouseLeave = this.onMouseLeave.bind(this);
+  }
+
+  componentDidMount() {
+    this.tooltip = this.refs.tooltipPosition;
+  }
+
+  onMouseEnter(item) {
+    this.renderTooltip(item);
+  }
+
+  onMouseLeave() {
+    this.tooltip.style.display = 'none';
+    this.tooltip.style.opacity = 1;
+  }
+
+  getTooltipDimension() {
+    const tooltip = document.getElementById('tooltip');
+
+    let lx;
+    let ly;
+    let element = tooltip;
+    for (lx = 0, ly = 0;
+      element != null;
+      lx += element.offsetLeft,
+      ly += element.offsetTop,
+      element = element.offsetParent);
+
+    return {
+      height: tooltip.clientHeight,
+      posX: lx,
+      posY: ly,
+      width: tooltip.clientWidth,
+    };
+  }
+
+  renderTooltipLabel() {
+    return (
+      <Box roundedCorners>
+        <div id="tooltip" className="tooltip">
+          <Text>
+            {this.content}
+          </Text>
+        </div>
+      </Box>
+    );
+  }
+
+  renderTooltip(item) {
+    this.tooltip.style.display = 'inline-block';
+    this.tooltip.style.left = '0px';
+    this.tooltip.style.top = '0px';
+    this.tooltip.style.height = this.props.height;
+    this.tooltip.style.width = this.props.width;
+    let styles = 'tooltipPosition ';
```

Comment:
I don't think it is better to create an array and convert it to string while you just can have the string directly, we aren't replacing strings or segregating them or.., it is just concat so I don't see the danger. What it needs to be done will not get any benefit from doing it with array and it will just have less performance.